src/python/pants/task/task.py

```diff
@@ -220,6 +221,11 @@ def get_passthru_args(self):
         passthru_args.extend(self.context.options.passthru_args_for_scope(self.options_scope))
         return passthru_args
 
+    @property
+    def debug(self) -> bool:
```

Comment:
I disagree with this one. I think it's much nicer for call sites to say `if log_level.debug` than `if log_level.is_debug_enabled`. 

`is_debug_enabled` suggests to me that you could have multiple log levels enabled at the same time, which isn't true. It's an enum. You can only have one log level at any given time.

Reply:
I think putting that much trust in words is probably not a great idea either way since, as demonstrated, they will be interpreted differently after you've wasted alot of sweat. That said - there was precedent in `src/python/pants/backend/python/tasks/pytest_run.py` which had this property, but named `_debug`.

Reply:
Added a doc string that lies, but the lie is how people thought this worked. Everyone checked for `-ldebug == "debug"` previously, when they probably meant to check if debug logging was enabled - which would have meant checking `-ldebug in ("debug", "trace")`.

Reply:
> `is_debug_enabled suggests` to me that you could have multiple log levels enabled at the same time, which isn't true.

It actually sort of is. Conceptually, -ltrace enables levels: trace, debug, info, warn and error. By type, yeah 1 level is set as the threshold.